peakinvestigator/lib/uploader.py

```diff
@@ -0,0 +1,94 @@
+## -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016, Veritomyx, Inc.
+#
+# This file is part of the Python SDK for PeakInvestigator
+# (http://veritomyx.com) and is distributed under the terms
+# of the BSD 3-Clause license.
+from peakinvestigator.lib import *
+
+class Uploader(object):
+
+    def __init__(self, host, token, progress_factory, log_level=logging.WARNING):
+        self.logger = logging.getLogger('Uploader')
+        self.logger.setLevel(log_level)
+        self.progress_factory = progress_factory
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.ssl_sock = ssl.wrap_socket(self.sock)
+        server_address = (socket.gethostbyname(host), 9999)
+
+        self.ssl_sock.connect(server_address)
+        self.ssl_sock.sendall(token.encode('utf-8'))
+        response = self.ssl_sock.recv(4096)
+        self.logger.info('Server response:', response)
+
+        if response != 'Authorized'.encode('utf-8'):
+            raise Exception('Client not authorized')
+
+        self.buffer = bytearray(4096)
+
+        global Progress
+        Progress = eval(progress_factory)
```

Comment:
Interesting idea. What happens if the "progress factory" takes arguments you didn't expect? For example, what if the user of the SDK wants a progress that should send an email when the upload is finished?